embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/session/orchestrator/SessionOrchestratorImpl.kt

```diff
@@ -22,24 +39,24 @@ internal class SessionOrchestratorImpl(
             // the session service will not be registered to the activity listener and will not
             // start the cold session.
             // If so, force a cold session start.
-            sessionService.startSessionWithState(true, clock.now())
+            sessionService?.startSessionWithState(true, clock.now())
         }
     }
 
     override fun onForeground(coldStart: Boolean, startupTime: Long, timestamp: Long) {
         backgroundActivityService?.endBackgroundActivityWithState(timestamp)
```

Comment:
I assume that starting and ending a BA will just work even if the instantiation is done outside the init block? We may want to add a test to verify this

Reply:
The pre-existing behavior for both sessions/BA is if something was started & then disabled, it gets sent. If it was disabled and then an end attempt is made, nothing is recorded. I added some unit tests to simulate this.